### PR TITLE
Adding CI publish workflow

### PR DESCRIPTION
**Pull request by wallentx**
This adds CI to publish the vsix extension to the vscode marketplace

### CODE REVIEW

**Comment by wallentx** [comment]
I tested all of this in `act` locally

**Comment by wallentx** [comment]
I made a pre-release targeting the commit hash of my branch:
https://marketplace.visualstudio.com/items?itemName=ChiaNetwork.chialisp-lsp-client

Note that it will only show `pre-release` if viewing from the actual VSCode App:


This does not show up in VSCodium by default - https://github.com/VSCodium/vscodium/blob/master/DOCS.md#how-to-use-the-vs-code-marketplace
so we might want to investigate publishing this to open-vsx as well. I did verify that you can download the .vsix file and install it in VSCodium.

I'll add uploading of the artifact to the release page in a separate PR.

File: .github/actions/setJobEnv.sh
Comment (cmmarslender) [inline .github/actions/setJobEnv.sh:1]: @wallentx if you're going to try and get this adopted among multiple repos, we should make it an action in https://github.com/Chia-Network/actions

File: .github/workflows/publish.yml
Comment (cmmarslender) [inline .github/workflows/publish.yml:12]:
```suggestion
        uses: actions/checkout@v3
```

doesn't v3 get latest from that series?

File: .github/workflows/publish.yml
Comment (cmmarslender) [inline .github/workflows/publish.yml:17]:
```suggestion
        uses: actions/setup-node@v3
```

Same comment as above

**Review by cmmarslender — DISMISSED** [review]
In addition to the changes noted inline, this repo also has the incorrect license (we usually use Apache 2.0) and its also giving credit to "Chia Inc" which is not us - this needs to be resolved before release.

License is messed up on the vscode website too now https://marketplace.visualstudio.com/items/ChiaNetwork.chialisp-lsp-client/license

Hopefully that will be resolved by publishing a new version - would have preferred we had all of this reviewed before publishing anything to the public, especially since the license issue was called out yesterday.

File: .github/workflows/publish.yml
Comment (wallentx) [reply .github/workflows/publish.yml]: Done!

**Comment by prozacchiwawa** [comment]
clvm_tools_rs is now in a publishable state.
I added a build of it in here in its wasm personality, which is pulled into runner via npm link.
When we pop clvm_tools_rs into npm, we can remove the ad-hoc build and add clvm_tools_rs to its dependencies properly.
I am unsure whether we want to release here or wait until the language server branch is merged, so being conservative on that.
In any case, while heavy handed, vsce package is fixed in here, assuming everything runs to completion (excluding missing system dependencies etc that i might have missed).

File: .github/actions/setJobEnv.sh
Comment (wallentx) [reply .github/actions/setJobEnv.sh]: done!

**Review by nirajpathak13 — APPROVED** [review]

**Review by cameroncooper — APPROVED** [review]

**Review by justinengland — APPROVED** [review]

**Pull request merged**
